diesel/src/sql_types/mod.rs

```diff
@@ -395,11 +395,110 @@ pub struct Timestamp;
 /// [`ToSql`]: /serialize/trait.ToSql.html
 /// [`FromSql`]: /deserialize/trait.FromSql.html
 /// [`serde_json::Value`]: /../serde_json/value/enum.Value.html
+#[cfg(all(
+    any(
+        feature = "postgres_backend",
+        feature = "sqlite",
+        feature = "mysql_backend"
+    ),
+    feature = "serde_json"
+))]
 #[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
 #[diesel(postgres_type(oid = 114, array_oid = 199))]
 #[diesel(mysql_type(name = "String"))]
+#[diesel(sqlite_type(name = "Text"))]
 pub struct Json;
 
+/// The [`jsonb`] SQL type.  This type can only be used with `feature =
+/// "serde_json"`
+///
+/// `jsonb` offers [several advantages][adv] over regular JSON:
+///
+/// > There are two JSON data types: `json` and `jsonb`. They accept almost
+/// > identical sets of values as input. The major practical difference
+/// > is one of efficiency. The `json` data type stores an exact copy of
+/// > the input text, which processing functions must reparse on each
+/// > execution; while `jsonb` data is stored in a decomposed binary format
+/// > that makes it slightly slower to input due to added conversion
+/// > overhead, but significantly faster to process, since no reparsing
+/// > is needed. `jsonb` also supports indexing, which can be a significant
+/// > advantage.
+/// >
+/// > ...In general, most applications should prefer to store JSON data as
+/// > `jsonb`, unless there are quite specialized needs, such as legacy
+/// > assumptions about ordering of object keys.
+///
+/// [adv]: https://www.postgresql.org/docs/current/static/datatype-json.html
+///
+/// ### [`ToSql`] impls
+///
+/// - [`serde_json::Value`]
+///
+/// ### [`FromSql`] impls
+///
+/// - [`serde_json::Value`]
+///
+/// [`ToSql`]: crate::serialize::ToSql
+/// [`FromSql`]: crate::deserialize::FromSql
+/// [`jsonb`]: https://www.postgresql.org/docs/current/datatype-json.html
+#[cfg_attr(
+    feature = "serde_json",
+    doc = "[`serde_json::Value`]: serde_json::value::Value"
+)]
+#[cfg_attr(
+    not(feature = "serde_json"),
+    doc = "[`serde_json::Value`]: https://docs.rs/serde_json/1.0.64/serde_json/value/enum.Value.html"
+)]
+///
+/// # Examples
+///
+/// ```rust
```

Comment:
@weiznich AFAIK this is the part that's not compatible with SQLite:

```
/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
/// #     use diesel::insert_into;
/// #     use self::contacts::dsl::*;
/// #     let connection = &mut connection_no_data();
/// #     diesel::sql_query("CREATE TABLE contacts (
/// #         id SERIAL PRIMARY KEY,
/// #         name VARCHAR NOT NULL,
/// #         address JSONB NOT NULL
/// #     )").execute(connection)?;
```

**But** it doesn't show in the rendered doc example:

```rust
table! {
    contacts {
        id -> Integer,
        name -> VarChar,
        address -> Jsonb,
    }
}

let santas_address: serde_json::Value = serde_json::from_str(r#"{
    "street": "Article Circle Expressway 1",
    "city": "North Pole",
    "postcode": "99705",
    "state": "Alaska"
}"#)?;
let inserted_address = insert_into(contacts)
    .values((name.eq("Claus"), address.eq(&santas_address)))
    .returning(address)
    .get_result::<serde_json::Value>(connection)?;
assert_eq!(santas_address, inserted_address);
```

And I think that code samples works fine for both SQLite and Pg.

Reply:
The main issue here is that this code is run as test for both the sqlite and the postgres backend. That means it will fail due to the not shown part for sqlite. We somehow need to fix that, possibly by conditionally creating the right table structure for both supported backends.

Reply:
@weiznich Done in 3a1a612c